### DISCUSSION TITLE
INCLUDE order

### DISCUSSION THREAD

**Discussion by Mogaba** [ltsp/ltsp#593] [Q&A] [answered]
In ltsp.conf, does INCLUDE order defines the real order of commands execution?
E.g. 
```
[clients]
INCLUDE=first
INCLUDE=second

[first]
POST_INIT_MAKE_DIR="mkdir /usr/mydir"

[second]
POST_INIT_ACCESS_DIR="ls /usr/mydir"
```

In the [second] section, will /usr/mydir always exist?

**Comment by alkisg** [comment, marked as answer]
Yes to the first question, no to the second! It's a bit more complicated, yet completely logical:

INCLUDEs are translated into shell function calls, so yes, they're always guaranteed to be called sequentially.
In your example, this will mean that POST_INIT_MAKE_DIR will be set first, and POST_INIT_ACCESS_DIR will be set second.

But note that I wrote "will be set", not "will be executed". By definition, the POST_x directives are executed at the **end** of the `ltsp init` execution. At that point, we don't care about which order they were set in. We only care about their names; they're executed in alphabetical order.

So, you need to rename POST_INIT_ACCESS_DIR to something alphabetically later than POST_INIT_MAKE_DIR.

A similar example in shell:

```shell
# We set a few variables; POST_B is set last:
$ POST_A=1; POST_C=3; POST_B=2
# Then we list them; the last one is POST_C:
$ set | grep POST
POST_A
POST_B
POST_C
```
